### PR TITLE
Update YLPClient signing for API v3

### PR DESCRIPTION
**Pull request by SSheldon**
(Based off of #45 and #46)

API v3 uses OAuth2; signing is simpler in OAuth2 because you just must provide a token, but before we can have a fully-functioning client you have to request that token and receive it from the API.

This change modifies the `YLPClient` so that instead of instantiating it directly, you call an asynchronous authorize method; when that completes it gives you a `YLPClient`.

TDOAuth, the library we were using before, only supports OAuth1, but OAuth2 is simple enough to write your own basic implementation for. I looked at NXOAuth2, but it was crashing during my testing and seemed like overkill for our needs.

The interface exposed here may need some additions in the future (tracking expiry of access tokens, storing them for repeat use) but I believe this change gives us a minimal yet fully-functional foundation to build off.

### CODE REVIEW

File: Classes/Client/YLPClient.m [hunk 1]
```diff
@@ -35,18 +35,82 @@ - (NSURLRequest *)requestWithPath:(NSString *)path {
...
+
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:urlComponents.URL];
+    request.HTTPMethod = @"GET";
+    NSString *authHeader = [NSString stringWithFormat:@"Bearer %@", self.accessToken];
+    [request setValue:authHeader forHTTPHeaderField:@"Authorization"];
+
+    return request;
+}
+
++ (NSCharacterSet *)URLEncodeAllowedCharacters {
```
Comment (thibaudrobelain) [inline Classes/Client/YLPClient.m]: put this in a category for NSCharacterSet?

File: Classes/Client/YLPClient.m [hunk 2]
```diff
@@ -35,18 +35,82 @@ - (NSURLRequest *)requestWithPath:(NSString *)path {
...
-                          consumerKey:self.consumerKey
-                       consumerSecret:self.consumerSecret
-                          accessToken:self.token
-                          tokenSecret:self.tokenSecret];
+    NSURLComponents *urlComponents = [[NSURLComponents alloc] init];
+    urlComponents.scheme = @"https";
+    urlComponents.host = kYLPAPIHost;
+    urlComponents.path = path;
+
+    NSMutableArray *queryItems = [NSMutableArray array];
```
Comment (thibaudrobelain) [inline Classes/Client/YLPClient.m]: maybe worth a small helper method to build the queryItems

File: Classes/Client/YLPClient.m (hunk 1 above)
Comment (SSheldon) [inline Classes/Client/YLPClient.m]: Feels a bit like overkill to me, where I'll have to add an `@interface` and an `@implementation` and make the method be prefixed, all for something that I don't want to expose publicly and is just used in one place here. Let me know if you feel strongly about this and I can still make it happen!

**Comment by thibaudrobelain** [comment]
@SSheldon Shipit!

**Pull request merged**